ownCloud/Settings/Passcode/PasscodeViewController.swift

```diff
@@ -0,0 +1,187 @@
+    @IBOutlet weak var timeTryAgainMessageLabel: UILabel?
+
+    // MARK: - Buttons
+    @IBOutlet var numberButtons: [ThemeButton]?
+    @IBOutlet weak var cancelButton: ThemeButton?
+
+    // MARK: - Initalization view
+    init(hiddenOverlay: Bool) {
+        super.init(nibName: "PasscodeViewController", bundle: nil)
+        self.hiddenOverlay = hiddenOverlay
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override var preferredStatusBarStyle : UIStatusBarStyle {
+        return Theme.shared.activeCollection.statusBarStyle
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        Theme.shared.register(client: self)
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+
+        Theme.shared.unregister(client: self)
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        self.view.addSubview(self.overlayPasscodeView)
+        self.overlayPasscodeView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
+        self.overlayPasscodeView.loadOverlay(isHidden: self.hiddenOverlay!, viewToShow: self.view)
+
+        self.cancelButton?.titleLabel?.text = "Cancel".localized
+
+        self.setIdentifiers()
+    }
+
+    func setEnableNumberButtons(isEnable: Bool) {
+
+        var alpha: CGFloat = 0.5
+
+        if isEnable {
+            alpha = 1.0
+        }
+
+        for button in self.numberButtons! {
+            button.isEnabled = isEnable
+            button.alpha = alpha
+        }
+    }
+
+    // MARK: - Actions
+
+    @IBAction func cancelButton(sender: UIButton) {
+        PasscodeManager.shared.dismissPasscode(animated: true)
+    }
+
+    @IBAction func numberButton(sender: UIButton) {
+        if let passcodeValue = self.passcodeValueTextField?.text {
+            self.passcodeValueTextField?.text = passcodeValue + String(sender.tag)
+        } else {
+            self.passcodeValueTextField?.text = String(sender.tag)
+        }
+
+        self.passcodeValueChanged(passcodeValue: (self.passcodeValueTextField?.text)!)
+    }
+
+    // MARK: - Passcode Flow
+
+    private func passcodeValueChanged(passcodeValue: String) {
+        PasscodeManager.shared.passcodeValueHasChange(passcodeValue: passcodeValue)
+    }
+
+    // MARK: - Identifiers
+
+    private func setIdentifiers() {
+
+        self.overlayPasscodeView.accessibilityIdentifier = "overlayView"
+
+        self.messageLabel?.accessibilityIdentifier = "messageLabel"
+        self.errorMessageLabel?.accessibilityIdentifier = "errorMessageLabel"
+        self.passcodeValueTextField?.accessibilityIdentifier = "passcodeValueTextField"
+        self.timeTryAgainMessageLabel?.accessibilityIdentifier = "timeTryAgainMessageLabel"
+
+        for button in self.numberButtons! {
+            button.accessibilityIdentifier = "number" + String(button.tag) + "Button"
```

Comment:
In order to determine the `accessibilityIdentifier` of a button, I now need to open the XIB, find the button, find its tag value, then put this string together by hand.

On the other hand, storing the accessibilityIdentifier directly in the XIB, it's as simple as opening the XIB, find the button and copy and paste the identifier to wherever it's needed for UI testing.

Reply:
All the accessibility are now on the XIB